### ISSUE TITLE
BUG: Urlencoding of path

### ISSUE THREAD

**Issue by JeroTw** [MatsuriDayo/NekoBoxForAndroid#753]
## Describe the problem

Expected behavior: The path sent as is

Actual behavior: Path string urlencoded first, then send encoded (example ? = %3F)

## How to reproduce

1) Open nekobox and import a VLess/VMESS config
2) Change the server to http server with debug (python3 -m http.server 8080)
3) Set html injector, any domain, and path /?test=1 
4) See the logs. ? replaced as %3F
![image](https://github.com/user-attachments/assets/c9a70e80-0b0a-4b47-926d-0c5b9eb61300)

Provide helpful screenshots, videos, text descriptions, subscription links, etc.

## log

If you have logs, please upload them. Please see the detailed steps for exporting logs in the documentation.

**Comment by eekrain** [comment]
yes i have noticed this also, it would be useful for using query instead of path to differentiate what protocol/transport we want to use with reverse proxy frontend

**Comment by eekrain** [comment]
u can also reproduce it with haproxy, lets say
```
acl vless query -i -m sub vless
```
and haproxy still cant read it as a query, but its just a plain path
